### PR TITLE
feat: add support to toggle the MacOS traffic lights controls

### PR DESCRIPTION
**Pull request by ThomasKranitsas**
This PR adds the ability to toggle the native MacOS "traffic light" controls after the BrowserWindow was created.

It also includes:
- Updated documentation
- Unit tests

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
This feels as though it should live around `titleBarStyle` or the `setClosable` `setMinimizable` style apis.

Its looking ok so far but can you explain why a whole new macOS only API instead of augmenting or adding a setter for titleBarStyle?

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -1471,6 +1471,16 @@ Adds a window as a tab on this window, after the tab for the window instance.
 Adds a vibrancy effect to the browser window. Passing `null` or an empty string
 will remove the vibrancy effect on the window.
 
+#### `win.setWindowControlsVisibility(visible)` _macOS_
+
+* `visible` Boolean
+
+Sets whether the window controls buttons should be visible.
+
+#### `win.isWindowControlsVisible()` _macOS_
```
Comment (zeke) [inline docs/api/browser-window.md]: It's unfortunate that "is" refers to a single thing but "window controls" is plural. But I suppose `areWindowControlsVisible()` might also be strange. 🤔

File: spec/api-browser-window-spec.js [hunk 2]
```diff
@@ -753,6 +753,72 @@ describe('BrowserWindow module', () => {
     })
   })
 
+  describe('BrowserWindow.setWindowControlsVisibility()', () => {
+    /**
+     * FIXME: Skip the tests instead of using the `return` here.
+     * - `.skip()` called in the 'before' hook doesn't affect
+     *     nested `describe`s.
+     * - Bad support of async test cases
```
Comment (alexeykuzmin) [inline spec/api-browser-window-spec.js]: @ThomasKranitsas What it is supposed to mean?

File: spec/api-browser-window-spec.js [hunk 3]
```diff
@@ -753,6 +753,72 @@ describe('BrowserWindow module', () => {
...
+    // Checks that window controls visibility behaviour remains the same in simple fullscreen mode
+    // if user did not specified it explicitly.
+    describe('when fullscreen titlebar on', () => {
+      beforeEach(() => {
+        w.destroy()
+        w = new BrowserWindow({ fullscreenWindowTitle: true })
+      })
+
+      it('hides controls when enters simple fullscreen mode', () => {
+        // BUG in SimpleFullScreen implementation? It does not trigger enter/leave-full-screen events
```
Comment (alexeykuzmin) [inline spec/api-browser-window-spec.js]: Can you please create an issue if there is a problem with these events?
https://github.com/electron/electron/issues/new/choose

File: atom/browser/api/atom_api_top_level_window.cc [hunk 4]
```diff
@@ -780,6 +780,14 @@ v8::Local<v8::Value> TopLevelWindow::GetContentView() const {
     return v8::Local<v8::Value>::New(isolate(), content_view_);
 }
 
+void TopLevelWindow::SetWindowControlsVisibility(bool visible) {
+  window_->SetWindowControlsVisibility(visible);
+}
+
+bool TopLevelWindow::IsWindowControlsVisible() {
```
Comment (codebytere) [inline atom/browser/api/atom_api_top_level_window.cc]: (nit): `IsWindowControlsVisible` may be better as `AreWindowControlsVisible`

File: docs/api/browser-window.md (hunk 1 above)
Comment (codebytere) [reply docs/api/browser-window.md]: oh didn't see this until i got down here...i think either are technically fine but `areWindowControlsVisible()` is slightly more readable?

File: atom/browser/api/atom_api_top_level_window.cc (hunk 4 above)
Comment (ThomasKranitsas) [reply atom/browser/api/atom_api_top_level_window.cc]: Fixed in the latest commit

File: docs/api/browser-window.md (hunk 1 above)
Comment (ThomasKranitsas) [reply docs/api/browser-window.md]: Fixed in the latest commit

**Comment by poiru** [comment]
This PR has been superseded by #13546. Thanks for your work on this Thomas!

**Pull request closed without merging**